tests/correct.py

```diff
@@ -71,6 +71,34 @@ def modify_list(num_list):
             return f'List contains type that is not an integer at position {i}'
     return num_list
 
+
+def sum_of_diagonal(matrix):
+    """
+    The function should take a 2D list (matrix) as input, where each sublist represents a row of the matrix.
+    The output should be a number of the sum of the primary (top-left to bottom-right) diagonal.
```

Comment:
Can you add clear instructions on if the matrix needs to be square too ??

Reply:
There are comments below that assume the input matrix is square. I believe it is line 79.

Reply:
But can we have like another return value for cases when it is not square matrix and make sure the student checks if the matrix is square .. I couldn't see that logic in your solution and hence wanted to make sure